File: dice.py


```python
#!/usr/bin/env python3
# dice.py
import argparse
import json
import csv
import random
import sys
from collections import defaultdict
from datetime import datetime
try:
    from colorama import init, Fore, Style
    init(autoreset=True)
    COLOR = True
except ImportError:
    COLOR = False
    # заглушки
    class Fore:
        GREEN = CYAN = YELLOW = RED = BLUE = MAGENTA = WHITE = RESET = ''
    Style = Fore

class Dice:
    def __init__(self, sides=6):
        self.sides = sides

    def roll(self, count=1):
        return [random.randint(1, self.sides) for _ in range(count)]

def main():
    parser = argparse.ArgumentParser(description="Игральные кости D6")
    parser.add_argument("--roll", type=int, default=1, help="Количество костей в броске")
    parser.add_argument("--count", type=int, default=1, help="Количество бросков")
    parser.add_argument("--seed", type=int, help="Seed для генератора")
    parser.add_argument("--stats", action="store_true", help="Показать статистику")
    parser.add_argument("--export-json", help="Экспорт в JSON")
    parser.add_argument("--export-csv", help="Экспорт в CSV")
    args = parser.parse_args()

    if args.seed is not None:
        random.seed(args.seed)

    dice = Dice()
    all_results = []
    stats = defaultdict(int)
    total_sum = 0

    print(Fore.CYAN + f"Бросаем {args.roll} костей, {args.count} раз(а)" + Style.RESET_ALL)
    for i in range(args.count):
        values = dice.roll(args.roll)
        s = sum(values)
        all_results.append(values)
        total_sum += s
        for v in values:
            stats[v] += 1
        # вывод с цветом
        val_str = ', '.join(str(v) for v in values)
        print(Fore.GREEN + f"Бросок {i+1}: [{val_str}] сумма = {s}" + Style.RESET_ALL)

    if args.count > 1:
        print(Fore.YELLOW + f"Общая сумма всех бросков: {total_sum}, среднее: {total_sum/args.count:.2f}" + Style.RESET_ALL)

    if args.stats:
        print(Fore.MAGENTA + "\nСтатистика выпадений:" + Style.RESET_ALL)
        for side in range(1, 7):
            print(f"  {side}: {stats.get(side, 0)}")

    if args.export_json:
        data = {
            "timestamp": datetime.now().isoformat(),
            "roll_count": args.roll,
            "num_rolls": args.count,
            "results": all_results,
            "statistics": dict(stats),
            "total_sum": total_sum,
            "average": total_sum / args.count if args.count > 0 else 0
        }
        with open(args.export_json, 'w') as f:
            json.dump(data, f, indent=2)
        print(Fore.BLUE + f"Результаты экспортированы в {args.export_json}" + Style.RESET_ALL)

    if args.export_csv:
        with open(args.export_csv, 'w', newline='') as f:
            writer = csv.writer(f)
            writer.writerow(['roll_number', 'values', 'sum'])
            for i, vals in enumerate(all_results, 1):
                writer.writerow([i, ','.join(map(str, vals)), sum(vals)])
        print(Fore.BLUE + f"Результаты экспортированы в {args.export_csv}" + Style.RESET_ALL)

if __name__ == "__main__":
    main()

File: dice.js
#!/usr/bin/env node
// dice.js
const { program } = require('commander');
const fs = require('fs');
const chalk = require('chalk');

class Dice {
    constructor(sides = 6) {
        this.sides = sides;
    }

    roll(count = 1) {
        const results = [];
        for (let i = 0; i < count; i++) {
            results.push(Math.floor(Math.random() * this.sides) + 1);
        }
        return results;
    }
}

program
    .option('--roll <number>', 'Количество костей в броске', parseInt, 1)
    .option('--count <number>', 'Количество бросков', parseInt, 1)
    .option('--seed <number>', 'Seed для генератора', parseInt)
    .option('--stats', 'Показать статистику')
    .option('--export-json <file>', 'Экспорт в JSON')
    .option('--export-csv <file>', 'Экспорт в CSV')
    .parse(process.argv);

const opts = program.opts();

if (opts.seed !== undefined) {
    // Простая имитация seed: используем множитель для детерминированности
    // В реальном коде лучше использовать seedrandom
    let s = opts.seed;
    const seedRandom = () => {
        s = (s * 9301 + 49297) % 233280;
        return s / 233280;
    };
    // Заменяем Math.random
    const origRandom = Math.random;
    Math.random = seedRandom;
    // Восстановим после выполнения (но для простоты оставим)
}

const dice = new Dice();
const results = [];
const stats = {};
let totalSum = 0;

console.log(chalk.cyan(`Бросаем ${opts.roll} костей, ${opts.count} раз(а)`));
for (let i = 0; i < opts.count; i++) {
    const values = dice.roll(opts.roll);
    const sum = values.reduce((a, b) => a + b, 0);
    results.push(values);
    totalSum += sum;
    for (const v of values) {
        stats[v] = (stats[v] || 0) + 1;
    }
    console.log(chalk.green(`Бросок ${i+1}: [${values.join(', ')}] сумма = ${sum}`));
}

if (opts.count > 1) {
    console.log(chalk.yellow(`Общая сумма всех бросков: ${totalSum}, среднее: ${(totalSum / opts.count).toFixed(2)}`));
}

if (opts.stats) {
    console.log(chalk.magenta('\nСтатистика выпадений:'));
    for (let side = 1; side <= 6; side++) {
        console.log(`  ${side}: ${stats[side] || 0}`);
    }
}

if (opts.exportJson) {
    const data = {
        timestamp: new Date().toISOString(),
        roll_count: opts.roll,
        num_rolls: opts.count,
        results: results,
        statistics: stats,
        total_sum: totalSum,
        average: opts.count > 0 ? totalSum / opts.count : 0
    };
    fs.writeFileSync(opts.exportJson, JSON.stringify(data, null, 2));
    console.log(chalk.blue(`Результаты экспортированы в ${opts.exportJson}`));
}

if (opts.exportCsv) {
    let csv = 'roll_number,values,sum\n';
    results.forEach((vals, idx) => {
        csv += `${idx+1},"${vals.join(',')}",${vals.reduce((a,b)=>a+b,0)}\n`;
    });
    fs.writeFileSync(opts.exportCsv, csv);
    console.log(chalk.blue(`Результаты экспортированы в ${opts.exportCsv}`));
}
